### ISSUE TITLE
apsw install failes on Apple ARM64 (M1)

### ISSUE THREAD

**Issue by diekhans** [rogerbinns/apsw#353]
pip install apws gets the below error.  This is coming from    sqlite-autoconf-3380500, which gets the same error when
compiling stand-alone.  Switch to sqlite-autoconf-3390200.tar.gz, which does compile, should fix this issue

```
      writing manifest file '/private/var/folders/3k/fvyxn7p97976wmlmtkd7ltfr0000z8/T/pip-pip-egg-info-kewbe6rx/apsw.egg-info/SOURCES.txt'
        Getting the SQLite amalgamation
          Fetching https://sqlite.org/2022/sqlite-autoconf-3380500.tar.gz
          Length: 3033164  SHA1: e37f9ea8ac088c6f0aa3462f8e1522ccc58de824  MD5: d7d14ef5e1cf2ce974bdd62a9cee347f
          Checksums verified
          Running configure to work out SQLite compilation flags
      Invalid configuration `arm64-apple-darwin21.4.0': machine `arm64-apple' not recognized
      configure: error: /bin/sh ./config.sub arm64-apple-darwin21.4.0 failed
      error: [Errno 2] No such file or directory: 'Makefile'

```

**Comment by diekhans** [comment]
Looks like this is fixed in the tree, it just needs to be release to  pypi

**Comment by rogerbinns** [comment]
My guess is that xcode is not installed.  SQLite's configure works fine for me on m1 mac, but I do have xcode installed.

The current pypi release does not have arm macos binaries so it does have to be compiled to install (this was a limitation of the various github actions and platforms).

The next release has updated actions and github platforms, and does have arm macos.  I'm expecting the release in the next 48 hours or so.  There is a preview with arm macos at [test pypi](https://test.pypi.org/project/apsw/#files)

**Comment by diekhans** [comment]
Hi Roger,

xcode is installed and pip install of apsw from github works great

The problem is  that to compile sqlite3, pypi apsw fetches sqlite-autoconf-3380500.tar.gz, whose configure does not work on M1

```
  % pip install apsw
  ...
  Fetching https://sqlite.org/2022/sqlite-autoconf-3380500.tar.gz
  ...
  Invalid configuration `arm64-apple-darwin21.4.0': machine `arm64-apple' not recognized

  % wget -nv https://sqlite.org/2022/sqlite-autoconf-3380500.tar.gz
  % tar -xf sqlite-autoconf-3380500.tar.gz
  % cd sqlite-autoconf-3380500/
  % ./configure
  ...
  checking build system type... Invalid configuration `arm64-apple-darwin21.4.0': machine `arm64-apple' not recognized
  configure: error: /bin/sh ./config.sub arm64-apple-darwin21.4.0 failed
```
  sqlite-autoconf-3390200.tar.gz configure works on M1.

I will test again when the new release is out, in the mean time,
install from github works.

THANK YOU for the essential package!!!

**Comment by rogerbinns** [comment]
I just double checked again and `sqlite-autoconf-3380500.tar.gz` does run configure on m1 (Monterey,  current Xcode) just fine.  Perhaps it is because I also have Xcode command line tools installed?  I don't have wget  installed and do not have homebrew or similar either.  gcc is found as `/usr/bin/gcc` and compiles hello world.   SQLite 3.33 was current around the time of m1 macs being released originally so support would have been added back then, if needed.

Also some background on why configure is needed for the record:

For Windows, the sqlite3 source uses `#if SQLITE_OS_WIN` or similar because configure is not run.  For other platforms configure works out if various header files are available and sets flags like `-DHAVE_UNISTD_H=1`.  It also works out if various library calls are available like `localtime_r()`, otherwise SQLite has to work around the lack of thread safety in that call.  The most important is looking for the `usleep` function (sleep for fractions of a second).  If that is not detected then SQLite can only sleep for whole numbers of seconds.  SQLite sleeps when there is contention for a database, and usually contention lasts for fractions of a second.  When whole second sleeps are used, things go really slow ...

What is currently in github is what will be released.  It is currently [doing a build for test pypi](https://github.com/rogerbinns/apsw/actions/runs/2766539038).

Closing since there is nothing I can do in apsw, and the issue won't arise going forward.

Off topic: ~25 years ago you and I worked for the same employer futzing with a "tool command language" :)
